src/test/regress/expected/distributed_intermediate_results.out

```diff
@@ -150,10 +220,111 @@ SELECT count(*), sum(x) FROM
+      WHERE colocated_with_target.a = 77
+), all_rows AS (
+    (SELECT * FROM shard_1) UNION ALL (SELECT * FROM shard_2) UNION ALL
+    (SELECT * FROM shard_3) UNION ALL (SELECT * FROM shard_4)
+)
+SELECT count(*), sum(x) FROM all_rows;
+ count | sum  
+-------+------
+   100 | 4550
+(1 row)
+
+ROLLBACK;
+DROP TABLE source_table, target_table, colocated_with_target;
+--
+-- Case 3.
+-- range partitioning, text format, replication factor 2 (both source and destination)
+-- composite distribution column
+-- 
+-- only redistribute_task_list_results
+--
+CREATE TYPE composite_key_type AS (f1 int, f2 text);
+SET citus.shard_replication_factor TO 2;
+-- source
+CREATE TABLE source_table(key composite_key_type, value int, mapped_key composite_key_type);
+SELECT create_distributed_table('source_table', 'key', 'range');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CALL public.create_range_partitioned_shards('source_table', '{"(0,a)","(25,a)"}','{"(24,z)","(49,z)"}');
+INSERT INTO source_table VALUES ((0, 'a'), 1, (0, 'a'));  -- shard 1 -> shard 1
+INSERT INTO source_table VALUES ((1, 'b'), 2, (26, 'b')); -- shard 1 -> shard 2
+INSERT INTO source_table VALUES ((2, 'c'), 3, (3, 'c'));  -- shard 1 -> shard 1
+INSERT INTO source_table VALUES ((4, 'd'), 4, (27, 'd')); -- shard 1 -> shard 2
+INSERT INTO source_table VALUES ((30, 'e'), 5, (30, 'e')); -- shard 2 -> shard 2
+INSERT INTO source_table VALUES ((31, 'f'), 6, (31, 'f')); -- shard 2 -> shard 2 
+INSERT INTO source_table VALUES ((32, 'g'), 7, (8, 'g'));  -- shard 2 -> shard 1
+-- target
+CREATE TABLE target_table(key composite_key_type, value int);
+SELECT create_distributed_table('target_table', 'key', 'range');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CALL public.create_range_partitioned_shards('target_table', '{"(0,a)","(25,a)"}','{"(24,z)","(49,z)"}');
+-- colocated with target, used for routing calls to read_intermediate_results
+CREATE TABLE colocated_with_target(key composite_key_type);
+SELECT create_distributed_table('colocated_with_target', 'key', 'range');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CALL public.create_range_partitioned_shards('colocated_with_target', '{"(0,a)","(25,a)"}','{"(24,z)","(49,z)"}');
+-- one value per shard, so we can route calls to read_intermediate_shards
+INSERT INTO colocated_with_target VALUES ((0,'a')), ((25, 'a'));
+BEGIN;
+CREATE TABLE distributed_result_info AS
+  SELECT * FROM redistribute_task_list_results('test', $$ SELECT mapped_key, value FROM source_table $$, 'target_table');
+SELECT * FROM distributed_result_info ORDER BY shardid;
+ shardid |                                       colocated_results                                       
+---------+-----------------------------------------------------------------------------------------------
+ 4213602 | {test_from_4213600_to_0,test_from_4213600_to_0,test_from_4213601_to_0,test_from_4213601_to_0}
```

Comment:
1. this shouldn't have duplicates
2. log which transfers happened to make sure we handled replication correctly.